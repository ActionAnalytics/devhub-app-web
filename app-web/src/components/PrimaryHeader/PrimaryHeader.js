import React from 'react';
import PropTypes from 'prop-types';
import { Flag } from 'flag';
import classes from './PrimaryHeader.module.css';
import { APP_TITLE } from '../../constants/ariaLabels';
import Banner from '../UI/Banner/Banner';
import Hamburger from '../UI/Hamburger/Hamburger';
import Login from '../Auth/Login/Login';

export const PrimaryHeader = ({ showHamburger, hamburgerClicked }) => (
  <header className={classes.PrimaryHeader}>
    <Banner title={APP_TITLE} navigateOnClickPath="/" />
    <div className={classes.Other}>
      <Flag name="features.login">
        <Login />
      </Flag>
      {showHamburger ? (
        <Hamburger clicked={hamburgerClicked} className={classes.Hamburger} />
      ) : null}
    </div>
  </header>
);

PrimaryHeader.propTypes = {
  showHamburger: PropTypes.bool,
  hamburgerClicked: PropTypes.func,
};

PrimaryHeader.defaultProps = {
  showHamburger: false,
  hamburgerClicked: () => undefined,
};

export default PrimaryHeader;
