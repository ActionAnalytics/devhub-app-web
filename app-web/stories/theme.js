import { create } from '@storybook/theming';
import { DEVHUB_PALETTE } from '../theme';

export default create({
  base: 'light',

  colorPrimary: 'hotpink',
  colorSecondary: 'deepskyblue',

  // UI
  appBg: 'white',
  appContentBg: 'white',
  appBorderColor: 'grey',
  appBorderRadius: 4,

  // Typography
  fontBase: 'sans-serif',
  fontCode: 'monospace',

  // Text colors
  textColor: DEVHUB_PALETTE.darkgrey,
  textInverseColor: 'rgba(255,255,255,0.9)',

  // Toolbar default and active colors
  barTextColor: DEVHUB_PALETTE.white,
  barSelectedColor: 'black',
  barBg: DEVHUB_PALETTE.blue,

  // Form colors
  inputBg: 'white',
  inputBorder: 'silver',
  inputTextColor: 'black',
  inputBorderRadius: 4,

  brandTitle: 'My custom storybook',
  brandUrl: 'https://example.com',
  brandImage: 'https://placehold.it/350x150',
});
