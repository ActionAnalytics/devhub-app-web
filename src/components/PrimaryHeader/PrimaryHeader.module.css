.PrimaryHeader {
    background-color: #036;
    border-bottom: 2px solid #fcba19;
    padding: 9px 0 10px;
    color: #fff;
    display: flex;
    align-items: center;
}