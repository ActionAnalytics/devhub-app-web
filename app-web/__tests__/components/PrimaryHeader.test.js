import React from 'react';
import { render } from 'react-testing-library';
import { ThemeProvider } from 'emotion-theming';
import theme from '../../theme';
import { PrimaryHeader } from '../../src/components/PrimaryHeader/PrimaryHeader';

describe('Primary Header Component', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <PrimaryHeader />
      </ThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
