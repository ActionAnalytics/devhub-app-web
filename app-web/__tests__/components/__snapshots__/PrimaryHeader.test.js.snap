// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary Header Component it matches snapshot 1`] = `
<Header>
  <Banner />
  <Panel>
    <LoginWrapper>
      <Login
        authenticated={false}
      />
    </LoginWrapper>
  </Panel>
</Header>
`;
